src/node/transaction.cpp

```diff
@@ -51,7 +51,11 @@ TransactionError BroadcastTransaction(NodeContext& node, const CTransactionRef t
         if (!existingCoin.IsSpent()) return TransactionError::ALREADY_IN_CHAIN;
     }
     if (!node.mempool->exists(hashTx)) {
-        // Transaction is not already in the mempool.
+        // A transaction with the same txid is not already in the mempool. Note that it is still
+        // possible for a transaction with same nonwitness data but different witness to be in the
+        // mempool - in that case, we treat them as the same transaction and don't try to submit tx
+        // to mempool. This logic must be changed if we allow replacements of
+        // same-txid-different-wtxid transactions in the future.
```

Comment:
I find this comment quite confusing. I initially read the "Note that it is still possible for a transaction with same nonwitness data but different witness to be in the mempool." as meaning "it is possible for this condition to be true in this code branch".

Does this seem better to you:

```diff
-    if (!node.mempool->exists(hashTx)) {
-        // A transaction with the same txid is not already in the mempool. Note that it is still
-        // possible for a transaction with same nonwitness data but different witness to be in the
-        // mempool - in that case, we treat them as the same transaction and don't try to submit tx
-        // to mempool. This logic must be changed if we allow replacements of
-        // same-txid-different-wtxid transactions in the future.
+    if (node.mempool->exists(hashTx)) {
+        // There's already a transaction in the mempool with this txid. Don't try to submit
+        // this transaction to the mempool, but do attempt to reannounce it if relay=true.
+        // The mempool transaction may have the same or different witness (and wtxid) as this
+        // transaction.
+    } else {
```